doctr/datasets/loaders/detection/core.py

```diff
@@ -106,42 +102,46 @@ def __data_generation(
                 polys, polys_mask = [], []
 
             image = cv2.imread(os.path.join(self.images_path, image_name))
+            image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
             h, w = image.shape[:2]
 
             # Initialize mask and gt
             gt = np.zeros((h, w), dtype=np.float32)
             mask = np.ones((h, w), dtype=np.float32)
 
             # Draw each polygon on gt
-            for poly in polys:
+            for poly, to_mask in zip(polys, to_masks):
                 poly = np.array(poly)
+                if to_mask is True:
+                    cv2.fillPoly(mask, poly.astype(np.int32)[np.newaxis, :, :], 0)
+                    continue
                 height = max(poly[:, 1]) - min(poly[:, 1])
                 width = max(poly[:, 0]) - min(poly[:, 0])
-                # generate true and mask
                 if min(height, width) < self.min_size_box:
                     cv2.fillPoly(mask, poly.astype(np.int32)[np.newaxis, :, :], 0)
                     continue
                 cv2.fillPoly(gt, [poly.astype(np.int32)], 1)
 
-            # Fill mask
-            for poly in polys_mask:
-                poly = np.array(poly)
-                cv2.fillPoly(mask, poly.astype(np.int32)[np.newaxis, :, :], 0)
-
-            # Resize and batch
+            # Cast
             image = tf.cast(image, tf.float32)
+            gt = tf.cast(gt, tf.float32)
+            mask = tf.cast(mask, tf.float32)
+            # Resize
             image = tf.image.resize(image, [*self.input_size], method='bilinear')
-            gt = resize(gt, self.input_size)
-            mask = resize(mask, self.input_size)
-
+            gt = tf.image.resize(gt, [*self.input_size], method='bilinear')
+            mask = tf.image.resize(mask, [*self.input_size], method='bilinear')
+            # Batch
             batch_images[index, ] = image
             batch_gts[index, ] = gt
             batch_masks[index, ] = mask
 
-        # Normalize
+        # Stack batches
         batch_images = tf.stack(batch_images, axis=0)
-        std = tf.cast((0.287, 0.274, 0.264), tf.float32)
-        mean = tf.cast((0.772, 0.785, 0.798), tf.float32)
+        batch_gts = tf.stack(batch_gts, axis=0)
+        batch_masks = tf.stack(batch_masks, axis=0)
+        # Normalize
+        std = tf.cast(self.std_rgb, tf.float32)
+        mean = tf.cast(self.mean_rgb, tf.float32)
```

Comment:
You can cast them into TF tensors in the constructors since this doesn't change afterwards